fiftyone/utils/annotations.py

```diff
@@ -101,30 +108,126 @@ def draw_labeled_image(
     anno_img = etaa.annotate_image(
-        img, image_labels, annotation_config=annotation_config
+        img, frame_labels, annotation_config=annotation_config
     )
 
     etai.write(anno_img, outpath)
 
 
-def _get_image_label_fields(sample):
-    return [
-        f for f in sample.field_names if isinstance(sample[f], fol.ImageLabel)
-    ]
+def draw_labeled_videos(
+    samples, anno_dir, label_fields=None, annotation_config=None
+):
+    """Renders annotated versions of the video samples with label field(s)
+    overlaid to the given directory.
+
+    The filenames of the sample videos are maintained, unless a name conflict
+    would occur in ``anno_dir``, in which case an index of the form
+    ``"-%d" % count`` is appended to the base filename.
+
+    The videos are written in format ``fo.config.default_video_ext``.
+
+    Args:
+        samples: an iterable of :class:`fiftyone.core.sample.Sample` instances
+        anno_dir: the directory to write the annotated videos
+        label_fields (None): a list of :class:`fiftyone.core.labels.ImageLabel`
+            fields on the frames of the samples to render. If omitted, all
+            compatiable fields are rendered
+        annotation_config (None): an :class:`AnnotationConfig` specifying how
+            to render the annotations
+
+    Returns:
+        the list of paths to the labeled videos
+    """
+    if annotation_config is None:
+        annotation_config = AnnotationConfig.default()
+
+    filename_maker = fou.UniqueFilenameMaker(output_dir=anno_dir)
+    output_ext = fo.config.default_video_ext
+
+    try:
+        num_samples = len(samples)
+    except:
+        num_samples = None
+
+    outpaths = []
+
+    for idx, sample in enumerate(samples, 1):
+        outpath = filename_maker.get_output_path(
+            sample.filepath, output_ext=output_ext
+        )
+
+        if num_samples is not None:
+            logger.info(
+                "Rendering video %d/%d: '%s'", idx, num_samples, outpath
+            )
+        else:
+            logger.info("Rendering video %d: '%s'", idx, outpath)
+
+        draw_labeled_video(
+            sample,
+            outpath,
+            label_fields=label_fields,
+            annotation_config=annotation_config,
+        )
+        outpaths.append(outpath)
+
+    return outpaths
+
+
+def draw_labeled_video(
+    sample, outpath, label_fields=None, annotation_config=None
+):
+    """Draws an annotated version of the video sample with its label field(s)
+    overlaid to disk.
+
+    Args:
+        sample: a :class:`fiftyone.core.sample.Sample` instance
+        outpath: the path to write the annotated image
+        label_fields (None): a list of :class:`fiftyone.core.labels.ImageLabel`
+            fields on the frames of the sample to render. If omitted, all
+            compatiable fields are rendered
+        annotation_config (None): an :class:`AnnotationConfig` specifying how
+            to render the annotations
+    """
```

Comment:
Maybe log info for individual video draws? It could be unclear what is happening when used in a script.

Screenshot of the output from the description code snippet:
![Screenshot from 2020-10-08 21-36-18](https://user-images.githubusercontent.com/19821840/95531407-69dd4380-09ae-11eb-94a9-4824a8565023.png)

Reply:
Yeah I see what you mean. I've been trying to minimize compulsory logging and instead let the user add their own logging. IE, if they wanted to indicate that a single export was occurring, they can add their own logging statement to the script. And if they want very minimal logging, they set `show_progress_bars = False` system-wide and then (almost) nothing will be logged unless they explicitly do it.

Loading zoo datasets are a notable exception to this rule; there are some things that are always logged. 🤷‍♂️ 

Also, for things like rendering multiple videos, it felt necessary to include some intermediate progress logs so that the user could make sense of what was happening.